src/proxy.rs

```diff
@@ -0,0 +1,90 @@
+use std::sync::Arc;
+
+use anyhow::Error;
+use hyper::{
+    body::Bytes,
+    header::{AUTHORIZATION, WWW_AUTHENTICATE},
+    Body, Method, Request, Response, StatusCode,
+};
+use tokio::stream::StreamExt;
+
+use crate::client::{RpcError, RpcResponse};
+use crate::env::Env;
+
+pub async fn proxy_request(env: Arc<Env>, request: Request<Body>) -> Result<Response<Body>, Error> {
+    let (parts, body) = request.into_parts();
+    if parts.uri.path() == "/" || parts.uri.path() == "" || parts.uri.path().starts_with("/wallet/")
+    {
+        if parts.method == Method::POST {
+            let env_local = env.clone();
+            if let Some((name, user)) = parts
+                .headers
+                .get(AUTHORIZATION)
+                .and_then(|auth| env_local.users.get(auth))
+            {
+                let body_data = body.collect::<Result<Bytes, _>>().await?;
+                match serde_json::from_slice(body_data.as_ref()) {
+                    Ok(req) => {
+                        let env_local = env.clone();
+                        let name_local = Arc::new(name);
+                        let response = env
+                            .rpc_client
+                            .send(parts.uri.path(), &req, move |_path, req| {
+                                use futures::TryFutureExt;
+                                let name_local_ok = name_local.clone();
+                                let name_local_err = name_local.clone();
+                                let env_local_ok = env_local.clone();
+                                let env_local_err = env_local.clone();
+                                user.intercept(env_local.clone(), req)
+                                    .map_ok(move |res| {
+                                        if res.is_some() {
+                                            info!(
+                                                env_local_ok.logger,
+                                                "{} called {}: INTERCEPTED",
+                                                name_local_ok,
+                                                req.method.0
+                                            )
+                                        } else {
+                                            info!(
```

Comment:
These two should be debug to avoid huge amount of messages.

Reply:
I thought `slog` compiled out log levels of `debug` and below in release builds? I have found these log statements incredibly useful, so I'd like to not lose them in release builds. If there is a simple way to override that behavior in `slog` I'd love to know/.

Reply:
Yes, it can be enabled by feature flag. Feel free to enable it (or I can do it) by default.